### ISSUE TITLE
Poetry breaks when poetry-dynamic-versioning-plugin is installed

### ISSUE THREAD

**Issue by tunayokumus** [mtkennerly/poetry-dynamic-versioning#80]
Hi, 
first kudos for the app. I think it's brilliant.
However I am experiencing major problems.
OS: Ubuntu 20.04 (on WSL2)
Poetry: version 1.2.0b1

Once I have the poetry-dynamic-versioning-plugin installed via `poetry plugin add` command, then poetry cannot run on folders without a pyproject.toml. Not even the `poetry init` or `poetry --help`

On a project folder with properly configured pyproject.toml, thing seems to work properly though. But also here the `poetry plugin show` command fails.

Once I delete the plugin, poetry is working fine again.

Here are the logs:

```
poetry init -vvv 
  Stack trace:

  8  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/cleo/application.py:330 in run
      328│ 
      329│             try:
    → 330│                 exit_code = self._run(io)
      331│             except Exception as e:
      332│                 if not self._catch_exceptions:

  7  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/poetry/console/application.py:172 in _run
      170│         self._disable_plugins = io.input.parameter_option("--no-plugins")
      171│ 
    → 172│         self._load_plugins(io)
      173│ 
      174│         return super()._run(io)

  6  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/poetry/console/application.py:320 in _load_plugins
      318│             manager = PluginManager("application.plugin")
      319│             manager.load_plugins()
    → 320│             manager.activate(self)
      321│ 
      322│         self._plugins_loaded = True

  5  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/poetry/plugins/plugin_manager.py:48 in activate
       46│     def activate(self, *args: Any, **kwargs: Any) -> None:
       47│         for plugin in self._plugins:
    →  48│             plugin.activate(*args, **kwargs)
       49│ 
       50│     def _load_plugin_entrypoint(self, entrypoint: entrypoints.EntryPoint) -> None:

  4  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/poetry_dynamic_versioning/plugin.py:96 in activate
       94│         )
       95│ 
    →  96│         config = _get_config(self._application.poetry.pyproject.data)
       97│         if not config["enable"]:
       98│             return

  3  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/poetry/console/application.py:119 in poetry
      117│             return self._poetry
      118│ 
    → 119│         self._poetry = Factory().create_poetry(
      120│             Path.cwd(), io=self._io, disable_plugins=self._disable_plugins
      121│         )

  2  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/poetry/factory.py:39 in create_poetry
       37│             io = NullIO()
       38│ 
    →  39│         base_poetry = super().create_poetry(cwd)
       40│ 
       41│         locker = Locker(

  1  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/poetry/core/factory.py:37 in create_poetry
       35│         from poetry.core.pyproject.toml import PyProjectTOML
       36│ 
    →  37│         poetry_file = self.locate(cwd)
       38│         local_config = PyProjectTOML(path=poetry_file).poetry_config
       39│ 

  RuntimeError

  Poetry could not find a pyproject.toml file in /home/user/some_dir/ or its parents

  at ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/poetry/core/factory.py:455 in locate
      451│             if poetry_file.exists():
      452│                 return poetry_file
      453│ 
      454│         else:
    → 455│             raise RuntimeError(
      456│                 f"Poetry could not find a pyproject.toml file in {cwd} or its parents"
      457│             )
      458│ 
```

```
poetry plugin show -vvv
Loading configuration file /home/myuser/.config/pypoetry/config.toml
Loading configuration file /home/myuser/.config/pypoetry/auth.toml


  Stack trace:

  7  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/cleo/application.py:330 in run
      328│ 
      329│             try:
    → 330│                 exit_code = self._run(io)
      331│             except Exception as e:
      332│                 if not self._catch_exceptions:

  6  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/poetry/console/application.py:174 in _run
      172│         self._load_plugins(io)
      173│ 
    → 174│         return super()._run(io)
      175│ 
      176│     def _configure_io(self, io: IO) -> None:

  5  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/cleo/application.py:425 in _run
      423│                 io.set_input(ArgvInput(argv))
      424│ 
    → 425│         exit_code = self._run_command(command, io)
      426│         self._running_command = None
      427│ 

  4  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/cleo/application.py:467 in _run_command
      465│ 
      466│         if error is not None:
    → 467│             raise error
      468│ 
      469│         return event.exit_code

  3  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/cleo/application.py:451 in _run_command
      449│ 
      450│             if event.command_should_run():
    → 451│                 exit_code = command.run(io)
      452│             else:
      453│                 exit_code = ConsoleCommandEvent.RETURN_CODE_DISABLED

  2  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/cleo/commands/base_command.py:118 in run
      116│         io.input.validate()
      117│ 
    → 118│         status_code = self.execute(io)
      119│ 
      120│         if status_code is None:

  1  ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/cleo/commands/command.py:85 in execute
       83│ 
       84│         try:
    →  85│             return self.handle()
       86│         except KeyboardInterrupt:
       87│             return 1

  KeyError

  'poetry-dynamic-versioning'

  at ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/poetry/console/commands/plugin/show.py:54 in handle
       50│             category = "plugins"
       51│             if issubclass(plugin, ApplicationPlugin):
       52│                 category = "application_plugins"
       53│ 
    →  54│             package = packages_by_name[canonicalize_name(entry_point.name)]
       55│             plugins[package.pretty_name]["package"] = package
       56│             plugins[package.pretty_name][category].append(entry_point)
       57│ 
       58│         for name, info in plugins.items():
```

**Comment by mtkennerly** [comment]
Thanks for reporting this. I'll get this sorted out ASAP.

**Comment by mtkennerly** [comment]
I've just released poetry-dynamic-versioning-plugin v0.3.2 to fix this.

**Comment by tunayokumus** [comment]
Perfect, thanks. Works fine now.
